src/main/java/de/dataelementhub/model/handler/importhandler/ImportHandler.java

```diff
@@ -0,0 +1,248 @@
+      CloseableDSLContext ctx, List<MultipartFile> files,
+      String importDirectory, int userId,
+      int importId, String timestamp) {
+    String destination = importDirectory + "/" + userId + "/" + timestamp;
+    new File(importDirectory + "/" + userId).mkdir();
+    new File(destination).mkdir();
+    for (MultipartFile file : files) {
+      Path fileNameAndPath = Paths.get(destination, file.getOriginalFilename());
+      try {
+        unzip(fileNameAndPath.toString(), destination);
+        allFilesInFolder(ctx, destination, importId);
+      } catch (Exception e) {
+        ctx.update(IMPORT)
+            .set(IMPORT.STATUS, ProcessStatus.INTERRUPTED)
+            .set(IMPORT.LABEL, e.getMessage())
+            .where(IMPORT.ID.eq(importId))
+            .execute();
+      }
+    }
+  }
+
+  /** Unzip to defined directory. */
+  public static void unzip(String source, String destination) throws ZipException {
+    ZipFile zipFile = new ZipFile(source);
+    zipFile.extractAll(destination);
+  }
+
+  /** list all files inside a defined directory. */
+  public static void allFilesInFolder(CloseableDSLContext ctx, String folder, int importId)
+      throws Exception {
+    File inputFolder = new File(folder);
+    File[] listOfFiles = inputFolder.listFiles();
+    for (File file : Objects.requireNonNull(listOfFiles)) {
+      if (file.isFile() && !file.getName().contains(".zip")) {
+        importType(ctx, file.getAbsolutePath(), importId);
+      }
+    }
+  }
+
+  /** detect file type (xml/json) then handle importing it. */
+  public static void importType(CloseableDSLContext ctx, String fileToImport,
+      int importId) throws Exception {
+    if (fileToImport.contains(".xml")) {
+      importXml(ctx, fileToImport, importId);
+    } else if (fileToImport.contains(".json")) {
+      importJson(ctx, fileToImport, importId);
+    }
+  }
+
+  /** handles importing xml file. */
+  public static void importXml(
+      CloseableDSLContext ctx, String fileToImport, int importId)
+      throws Exception {
+    File file = new File(fileToImport);
+    validateAgainstSchema(fileToImport);
+    JAXBContext jaxbContext = JAXBContext.newInstance(ImportDto.class);
+    Unmarshaller jaxbUnmarshaller = jaxbContext.createUnmarshaller();
+    ImportDto importDtoBody = (ImportDto) jaxbUnmarshaller.unmarshal(file);
+    saveElements(ctx, importDtoBody.getStagedElements(), importId);
+  }
+
+  /** handles importing json file. */
+  public static void importJson(
+      CloseableDSLContext ctx, String fileToImport, int importId)
+      throws Exception {
+    System.setProperty(
+        "javax.xml.bind.context.factory", "org.eclipse.persistence.jaxb.JAXBContextFactory");
+    validateAgainstSchema(fileToImport);
+    JAXBContext jaxbContext = JAXBContext.newInstance(ImportDto.class);
+    Unmarshaller jaxbUnmarshaller = jaxbContext.createUnmarshaller();
+    jaxbUnmarshaller.setProperty(JAXBContextProperties.MEDIA_TYPE, "application/json");
```

Comment:
There should be media type constants ready to use for application/json. Please use them.

Reply:
There was a reason for using JAXBContext here. I can't 100% remember why I used it but I think it was because JAXBContext accepts nested JSON

Reply:
Uh - I simply meant replacing the actual String "application/json" with [MediaType.APPLICATION_JSON_VALUE](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/http/MediaType.html#APPLICATION_JSON_VALUE)

And now that I see it - please also rename the package here from ../handler/importhandler to simply ../handler/import

Reply:
"application/json" is replaced now with MediaType.APPLICATION_JSON_VALUE but importhandler can not be renamed to import because import is not a valid package name